server/utils/prisma/prisma-client.ts

```diff
@@ -1,7 +1,67 @@
 import { PrismaClient } from "@prisma/client";
 
+// Create a base Prisma client
 const globalForPrisma = global as unknown as { prisma: PrismaClient };
+export const basePrisma = globalForPrisma.prisma || new PrismaClient({
+  datasources: {
+    db: {
+      url: process.env.POSTGRES_PRISMA_URL, // Use connection pooler
+    },
+  },
+});
 
-export const prisma = globalForPrisma.prisma || new PrismaClient();
+if (process.env.NODE_ENV !== "production") globalForPrisma.prisma = basePrisma;
 
-if (process.env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;
+// Function to create an authenticated Prisma client instance
+export function createAuthenticatedPrismaClient(userId?: string, role?: string) {
+  // User ID is required for authenticated client
+  // It should be extracted from the JWT token
+  if (!userId && !role) {
+    return basePrisma;
+  }
+
+  // Create a new client with extensions to handle authentication
+  const prisma = new PrismaClient({
```

Comment:
I think its not a good approach to create a new prisma client instance for each request. We should be creating a single instance and reusing it with extensions. Please see https://github.com/prisma/prisma-client-extensions/tree/main/row-level-security & https://www.prisma.io/docs/orm/more/help-and-troubleshooting/nextjs-help#best-practices-for-using-prisma-client-in-development